core-plugins/src/main/java/co/cask/hydrator/plugin/transform/ScriptTransform.java

```diff
@@ -119,6 +119,8 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws Ille
     // try evaluating the script to fail application creation if the script is invalid
     init(null);
 
+    Schema outputSchema = (schema == null) ? pipelineConfigurer.getStageConfigurer().getInputSchema() : schema;
```

Comment:
we call init before this, and that initializes schema. i can add a comment here.